pkg/operator/observe_config.go

```diff
@@ -74,30 +77,46 @@ func NewConfigObserver(
 // must be information that is logically "owned" by another component.
 func (c ConfigObserver) sync() error {
 
-	observedConfig := map[string]interface{}{}
-	var err error
-
-	for _, observer := range c.observers {
-		observedConfig, err = observer(c.kubeClient, &rest.Config{}, observedConfig)
-		if err != nil {
-			return err
-		}
-	}
-
 	operatorConfig, err := c.operatorConfigClient.KubeApiserverOperatorConfigs().Get("instance", metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
 
-	// don't worry about errors
+	// don't worry about errs
 	currentConfig := map[string]interface{}{}
 	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&currentConfig)
-	if reflect.DeepEqual(currentConfig, observedConfig) {
-		return nil
-	}
+	observedConfig := map[string]interface{}{}
+	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&observedConfig)
 
-	glog.Infof("writing updated observedConfig: %v", diff.ObjectDiff(operatorConfig.Spec.ObservedConfig.Object, observedConfig))
-	operatorConfig.Spec.ObservedConfig = runtime.RawExtension{Object: &unstructured.Unstructured{Object: observedConfig}}
+	var errs []error
+	for _, observer := range c.observers {
+		var currErrs []error
+		observedConfig, currErrs = observer(c.kubeClient, &rest.Config{}, observedConfig)
+		errs = append(errs, currErrs...)
+	}
+	if !reflect.DeepEqual(currentConfig, observedConfig) {
```

Comment:
@deads2k worth using the apimachinery equality.Semantic.DeepEqual helper here?

Reply:
> @deads2k worth using the apimachinery equality.Semantic.DeepEqual helper here?

no